django/db/models/fields/__init__.py

```diff
@@ -738,7 +738,7 @@ def contribute_to_class(self, cls, name, private_only=False):
             # if you have a classmethod and a field with the same name, then
             # such fields can't be deferred (we don't have a check for this).
             if not getattr(cls, self.attname, None):
-                setattr(cls, self.attname, DeferredAttribute(self.attname))
+                setattr(cls, self.attname, DeferredAttribute(self))
```

Comment:
I added this missing change to the first commit.

Reply:
Thank you.